symphony-ws/src/main/java/se/havochvatten/symphony/scenario/ScenarioSnapshot.java

```diff
@@ -101,7 +101,11 @@ public void setPolygon(JsonNode polygon) {
 
     public NormalizationOptions getNormalization() { return normalization; }
 
-    public JsonNode getChanges() { return changes; }
+    public ObjectMapper getMapper() { return mapper; }
+
+    public JsonNode getChanges() { return changes.get("baseChanges"); }
+
+    public JsonNode getChangesForReport() { return changes; }
```

Comment:
This is not a kind of typo but the naming is deliberately chosen for a specific reason.
We're looking at a little bit of a special case, that relates to how the `ScenarioSnapshot` persists the data from a scenario. The "changes" json structure retains _both_ the general changes to the `Scenario` (keyed by `"baseChanges"`) _and_ the `ScenarioArea`s respective changes as a separate keyed map structure.

`getChanges()` is used here to implement the interface `BandChangeEntity` (since this same commit) because it allows us to pass the class `ScenarioSnapshot` as the parameter called `alternateChangeSource` to the `apply` method in `ScenarioService`, here:
https://github.com/havochvatten/MSP-Symphony/blob/0b7a896c85566c1ee407d086bcfc89b751b7e0ca/symphony-ws/src/main/java/se/havochvatten/symphony/scenario/ScenarioService.java#L145-L150
Previously, `ScenarioSnapshot`s' `getChanges` method was used _only_ in `ReportService` to fetch the changes as they're rendered in the report.  
We still need to do that, but only for the purpose of rendering the report sheet, hence the naming `getChangesForReport()`.
I'll argue that this serves the internal consistency better than your proposed change in that `getChanges` in this implementation returns the same data structure that the corresponding  method on the `Scenario` class will.

In any case, we can not just rename this without having to do a lot of other changes to the recalculation procedure.

If you disagree, I don't mind continuing the discussion 😄

Reply:
I am just thinking of consistency in naming the getter and setter so they get and set the same variable. It would help if a noob like me follow everything.  An alternative could be to instead rename the **set**Changes() to setChangesForReport() (and prehaps renaming the variable). Is that an option and consistent with the rest of the code?

Reply:
I wouldn't think that's actually appropriate either, because I feel that may incorrectly lead to the assumption that we're providing changes only for the report. But as you'll see, the area changes in the object we're setting are accessed in the recalculation as well.
Maybe we've got a better case for renaming the getter for the composite object to something like `getAllChanges` or something like that.
Would you think that's an acceptable option?

Reply:
And maybe then, we could achieve the kind of consistency you advocate by also renaming the setter correspondingly (`setAllChanges`), which I don't love to be honest 😊, but it's acceptable and won't require changing much code.

Reply:
(yes, obviously @delayensmjukvara is my private acct)
The setter is also defined in the mentioned interface, which unfortunately means that we can't easily rename it. It is implemented by two other classes (`Scenario` and `ScenarioArea`), with `ScenarioSnapshot` being the outlier.

Reply:
Is this a clear explanation?
https://github.com/havochvatten/MSP-Symphony/blob/c7d5919151856a35b78072edfb18e2f0964d5eed/symphony-ws/src/main/java/se/havochvatten/symphony/scenario/ScenarioSnapshot.java#L109-L117